osu.Game/Overlays/Chat/ChatLine.cs

```diff
@@ -210,5 +208,32 @@ private void updateMessageContent()
             username.Text = $@"{message.Sender.Username}" + (senderHasBackground ? "" : ":");
             contentFlow.Text = message.Content;
         }
+
+        private class MessageSender : ClickableContainer, IHasContextMenu
+        {
+            private readonly User sender;
+
+            public MessageSender(User sender)
+            {
+                this.sender = sender;
+            }
+
+            [BackgroundDependencyLoader(true)]
+            private void load(UserProfileOverlay profile)
+            {
+                Action = () => profile?.ShowUser(sender);
+            }
+
+            public MenuItem[] ContextMenuItems => new OsuMenuItem[]
+            {
+                new OsuMenuItem("Spectate"),
+                new OsuMenuItem("View Profile", MenuItemType.Highlighted, Action.Invoke),
+                new OsuMenuItem("Start Chat"),
+                new OsuMenuItem("Add as Friend"),
+                new OsuMenuItem("Invite to"),
+                new OsuMenuItem("Report"),

```

Comment:
this should also be destructive. in general i'm against adding menu options to things that aren't implemented yet, though. you could just leave View Profile here for now and add the rest as they are implemented.